File: memory-domain/src/main/java/com/memory/dto/map/MapRequest.java
package com.memory.dto.map;

import com.memory.domain.map.Map;
import com.memory.domain.map.MapType;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;

public class MapRequest {

    @Getter
    public static class Create {
        @NotBlank(message = "지도 이름은 필수 입력값입니다.")
        private String name;

        private String description;

        @NotBlank(message = "주소는 필수 입력값입니다.")
        private String address;

        @NotBlank(message = "위도는 필수 입력값입니다.")
        private String latitude;

        @NotBlank(message = "경도는 필수 입력값입니다.")
        private String longitude;

        @NotNull(message = "지도 타입은 필수 입력값입니다.")
        private MapType mapType;

        public Create(String name, String description, String address, String latitude, String longitude, MapType mapType) {
            this.name = name;
            this.description = description;
            this.address = address;
            this.latitude = latitude;
            this.longitude = longitude;
            this.mapType = mapType;
        }

        public Map toEntity() {
            return new Map(name, description, address, latitude, longitude, mapType);
        }
    }
}

File: memory-api/src/main/kotlin/com/memory/dto/map/MapRequest.kt
package com.memory.dto.map

import com.memory.domain.map.Map
import com.memory.domain.map.MapType
import com.memory.domain.member.Member
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

class MapRequest {

    data class Create(

        @field:NotBlank(message = "이름은 필수 입력값입니다.")
        val name: String,

        val description: String?,

        @field:NotBlank(message = "주소는 필수 입력값입니다.")
        val address: String?,

        @field:NotBlank(message = "위도는 필수 입력값입니다.")
        val latitude: String?,

        @field:NotBlank(message = "경도는 필수 입력값입니다.")
        val longitude: String?,

        @field:NotNull(message = "지도 타입은 필수 입력값입니다.")
        val mapType: MapType
    ) {
        fun toEntity(member: Member): Map {
            return Map(name, description, address, latitude, longitude, mapType, member)
        }
    }
}